crates/rooch-framework/sources/auth_validator/ecdsa_k1_recoverable_validator.move

```diff
@@ -38,32 +38,24 @@ module rooch_framework::ecdsa_k1_recoverable_validator {
         // compare newly passed public key with ecdsa recoverable public key length to ensure it's compatible
         assert!(
             vector::length(&public_key) == V_ECDSA_RECOVERABLE_PUBKEY_LENGTH,
-            error::invalid_argument(EMalformedAuthenticationKey)
+            error::invalid_argument(EInvalidPublicKeyLength)
         );
 
-        // ensure that the ecdsa recoverable public key to address isn't matched with the ed25519 account address
+        // User can rotate the authentication key arbitrarily, so we do not need to check the new public key with the account address.
+        let authentication_key = public_key_to_authentication_key(public_key);
         let account_addr = signer::address_of(account);
-        let ecdsa_recoverable_addr = ecdsa_k1_recoverable_public_key_to_address(public_key);
-        assert!(
-            account_addr != ecdsa_recoverable_addr,
-            error::invalid_argument(EMalformedAccount)
-        );
+        rotate_authentication_key(ctx, account_addr, authentication_key);
+    }
 
-        // serialize the address to an auth key and rotate it by calling rotate_authentication_key
-        let ecdsa_k1_recoverable_authentication_key = moveos_std::bcs::to_bytes(&ecdsa_recoverable_addr);
-        account_authentication::rotate_authentication_key<EcdsaK1RecoverableValidator>(
-            ctx,
-            account_addr,
-            ecdsa_k1_recoverable_authentication_key
-        );
+    fun rotate_authentication_key(ctx: &mut StorageContext, account_addr: address, authentication_key: vector<u8>) {
+        account_authentication::rotate_authentication_key<EcdsaK1RecoverableValidator>(ctx, account_addr, authentication_key);
     }
 
     public entry fun remove_authentication_key_entry<EcdsaK1RecoverableValidator>(ctx: &mut StorageContext, account: &signer) {
```

Comment:
`EcdsaK1RecoverableValidator` is in the current module, not a generic type.

Reply:
Same as the rotate_authentication_key_entry function

Reply:
Fixed by https://github.com/rooch-network/rooch/pull/610/commits/b7e01c8095f3046369a319f621719a86aeecfe69 and #614.